katello/katello-change-hostname

```diff
@@ -251,10 +258,14 @@ scenario_answers = YAML.load_file("/etc/foreman-installer/scenarios.d/#{@options
 
 unless @foreman_proxy_content
   STDOUT.puts "\nUpdating default #{@proxy}"
-  default_capsule_id = hammer_cmd("capsule list | grep #{old_hostname} | awk '{ print $1 }'").chomp
-  # Incorrect error message is piped to /dev/null, can be removed when http://projects.theforeman.org/issues/18186 is fixed
-  # For the same reason, we accept exit code 65 here.
-  hammer_cmd("capsule update --id #{default_capsule_id} --url https://#{@new_hostname}:9090 --new-name #{@new_hostname} 2> /dev/null", [0, 65])
+  proxy_id = get_default_proxy_id
+  if proxy_id
+    # Incorrect error message is piped to /dev/null, can be removed when http://projects.theforeman.org/issues/18186 is fixed
+    # For the same reason, we accept exit code 65 here.
+    hammer_cmd("capsule update --id #{proxy_id} --url https://#{@new_hostname}:9090 --new-name #{@new_hostname} 2> /dev/null", [0, 65])
```

Comment:
I wonder if we are already touching this, should we also go for replacing the hostname in a saner way?

```ruby
require 'uri'
capsule_uri = capsule['URL']
capsule_uri = URI.parse(capsule_uri)
capsule_uri.host = @new_hostname
capsule['URL'] = capsule_uri.to_s
```

Reply:
Wouldn't `capsule` be the old hostname here?

Reply:
`capsule` is the result of `JSON.parse(hammer_cmd("--output json capsule info --name #{@old_hostname}"))`